firedrake/assemble.py

```diff
@@ -632,76 +655,30 @@ def _make_tensor(form, bcs, *, diagonal, mat_type, sub_mat_type, appctx,
         raise AssertionError
 
 
-class FormAssembler(abc.ABC):
-    """Abstract base class for assembling forms.
+class FormAssemblerParloopBuilder(abc.ABC):
+    """Abstract base class for making ParloopBuilders for assembly.
```

Comment:
The pattern here seems to be "builder class that constructs another builder class". That seems wrong. I am quite confused by the inheritance going on.

Reply:
This is no longer relevant.